posts/2020-11-27-typing-your-actors.adoc

```diff
@@ -0,0 +1,389 @@
+= Typing your actors
+Pedro Ferreira <pedro.ferreira@lunatech.nl>
+v1.0, 2020-11-27
+:title: Typing your actors step by step
+:tags: [akka, migration, akka-typed]
+
+== Background
+
+After some frustration with our project's choice for a planning poker solution, I did what any reasonable developer would do. No, I didn't search for a more suitable solution, I decided to write my own and do a blog post about it.
+
+The main idea is having a simple application that involves a bit of the three core Akka libraries:
+
+* Http: REST api for a Vue.js frontend
+* Streams: Websocket bi-directional data handling
+* Actors: Rooms management
+
+Initially I chose to use an untyped actor system intentionally, so we can actually talk about something that might be more useful, migrating your old Akka applications to a typed actor system. Why would you want to do that, well for me it is the type safety of not having everything being a function of `Any => Unit` and strong support to avoid anti-patterns (more eloquently explained https://www.youtube.com/watch?v=vF_-28I6yGE[here]), but feel free to search more if this migration is something for your project.
```

Comment:
```suggestion
Initially, I deliberatly chose to use a classic actor implementation so that I could actually also talk about something that might be more useful: migrating an Akka Classic application to an Akka Typed version.

"Why would you want to do that?", you may ask... Well, for me it is the type safety of Akka Typed which means that an Actor's behaviour isn't just a function of `Any => Unit` **_and_** strong support to avoid certain anti-patterns (more eloquently explained https://www.youtube.com/watch?v=vF_-28I6yGE[here]), but feel free to search more if this migration is something for your project.
```